### DISCUSSION TITLE
how to use Lux.jl utility 'BinaryCrossEntropy'

### DISCUSSION THREAD

**Discussion by CMoebus** [LuxDL/Lux.jl#842] [Q&A] [answered]
Hi,
I am am new to Lux.jl and want to use the 'BinaryCrossEntropy' utility according to the small example (https://lux.csail.mit.edu/stable/api/Lux/utilities#Lux.BinaryCrossEntropyLoss):

```julia
julia> bce = BinaryCrossEntropyLoss();
julia> y_bin = Bool[1, 0, 1];
julia> y_model = Float32[2, -1, pi]
3-element Vector{Float32}:
  2.0
 -1.0
  3.1415927
```

but my Pluto/Julia script:

```julia
begin 
	using Lux, LossFunctions
	LossFunctions.CrossEntropyLoss()                     # this was recognized
	bce = Lux.BinaryCrossEntropyLoss()                 # this not !
end # begin
```

led to this error:

```julia
 UndefVarError: `BinaryCrossEntropyLoss` not defined
    getproperty(::Module, ::Symbol)@Base.jl:31
    top-level scope@[Local: 4](http://localhost:1234/edit?id=7b7ed1be-57e9-11ef-3d33-2d8cf3743530#)
```

What was my fault ? Thanks for help, Claus

**Comment by avik-pal** [comment]
Can you share what is `pkgversion(Lux)`, I am guessing an old version of Lux was installed for some reason.

**Comment by CMoebus** [comment]
v"0.5.17"

https://uol.de/en/lcs/
________________________________

**Comment by CMoebus** [comment, marked as answer]
Hi, I solved the problem by myself ! I deleted the Julia registry and let it rebuild again. Then I updated the Lux package. Now, I have the latest version pkgversion(Lux) = 0.5.63. 'BinaryCrossEntropy' is recognized now and the example in the doc is running.
Thanks for your patience, Claus
